### PR TITLE
Adding resolvo-cpp

### PR DESCRIPTION
**Pull request by baszalmstra**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by baszalmstra** [comment]
@conda-forge/help-c-cpp, ready for review

File: recipes/resolvo-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+test:
+  commands:
+    - test -d ${PREFIX}/include/resolvo  # [unix]
+    - test -f ${PREFIX}/include/resolvo/resolvo.h  # [unix]
+    - test -f ${PREFIX}/include/resolvo/resolvo_internal.h  # [unix]
+    - test -e ${PREFIX}/lib/libresolvo_cpp${SHLIB_EXT}  # [unix]
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo_internal.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll.lib (exit 1)  # [win]
```
[the remark this answers is not in the document]
Comment (baszalmstra) [reply recipes/resolvo-cpp/meta.yaml:33]: Yeah we could turn this into a multi output recipe rhat generates both the static and the shared libraries. 

There are some complexities regarding having both packages available and potentially clobbering each other. Im not sure what the right approach is.

But! It is also very straightforward to statically link the binaries using Cmake directly, maybe that should be the preferred approach?

[the remark this answers is not in the document]
Comment (jjerphan) [reply recipes/resolvo-cpp/meta.yaml:33]: On our side, I guess we could for the first experiments use the shared library.

File: recipes/resolvo-cpp/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+cmake ${CMAKE_ARGS} -GNinja $SRC_DIR -DRust_CARGO_TARGET=${CARGO_BUILD_TARGET}
```
Comment (jjerphan) [inline recipes/resolvo-cpp/build.sh:5]: Should `Release` be used as the build type?

```suggestion
cmake \
      -GNinja $SRC_DIR \
      -DRust_CARGO_TARGET=${CARGO_BUILD_TARGET} \
      -DCMAKE_BUILD_TYPE=Release \
      ${CMAKE_ARGS}
```

File: recipes/resolvo-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{% set name = "resolvo-cpp" %}
+{% set version = "0.1.0" %}
```
Comment (jjerphan) [inline recipes/resolvo-cpp/meta.yaml:2]: Should this recipe [export runtime requirements](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements) to encode version range for ABI stability?

File: recipes/resolvo-cpp/bld.bat [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@echo on
+
+cmake %CMAKE_ARGS% -GNinja %SRC_DIR%
```
Comment (jjerphan) [inline recipes/resolvo-cpp/bld.bat:3]: Shouldn't `Rust_CARGO_TARGET` be set as well?
```suggestion
cmake %CMAKE_ARGS% -GNinja %SRC_DIR% -DRust_CARGO_TARGET=%CARGO_BUILD_TARGET%
```

Comment (jjerphan) [reply recipes/resolvo-cpp/bld.bat:3]: With the previous remark about the build type, the suggestion above gives:
```suggestion
cmake ^
      -GNinja %SRC_DIR% ^
      -DRust_CARGO_TARGET=%CARGO_BUILD_TARGET% ^
      -DCMAKE_BUILD_TYPE=Release ^
      %CMAKE_ARGS% 
```

File: recipes/resolvo-cpp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    - cmake
+    - ninja
+    - cargo-bundle-licenses
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/resolvo  # [unix]
+    - test -f ${PREFIX}/include/resolvo/resolvo.h  # [unix]
+    - test -f ${PREFIX}/include/resolvo/resolvo_internal.h  # [unix]
+    - test -e ${PREFIX}/lib/libresolvo_cpp${SHLIB_EXT}  # [unix]
```
Comment (jjerphan) [inline recipes/resolvo-cpp/meta.yaml:29]:
```suggestion
    - test -e ${PREFIX}/lib/libresolvo_cpp${SHLIB_EXT}  # [unix]
    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoTargets.cmake  # [unix]
    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoConfigVersion.cmake  # [unix]
    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoConfig.cmake  # [unix]
```

File: recipes/resolvo-cpp/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo_internal.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll.lib (exit 1)  # [win]
```
Comment (jjerphan) [inline recipes/resolvo-cpp/meta.yaml:33]:
```suggestion
    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll.lib (exit 1)  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoTargets.cmake  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfig.cmake  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfigVersion.cmake  # [win]
```

**Review by jjerphan — COMMENTED** [review]
On testing CMake installations.

File: recipes/resolvo-cpp/build.sh (hunk 2 above)
Comment (baszalmstra) [reply recipes/resolvo-cpp/build.sh:5]: I am pretty sure this is already included in the `CMAKE_ARGS`

File: recipes/resolvo-cpp/bld.bat (hunk 4 above)
Comment (baszalmstra) [reply recipes/resolvo-cpp/bld.bat:3]: `-DRust_CARGO_TARGET=%CARGO_BUILD_TARGET%` is only necesary when cross compiling which we arent currently doing on windows. But I will add it for future self.

File: recipes/resolvo-cpp/meta.yaml (hunk 3 above)
Comment (baszalmstra) [reply recipes/resolvo-cpp/meta.yaml:2]: Yes, good catch! I added it.

File: recipes/resolvo-cpp/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (baszalmstra) [reply recipes/resolvo-cpp/meta.yaml]: Yes, lets do that in a later iteration.

File: recipes/resolvo-cpp/bld.bat [hunk 7]
```diff
@@ -0,0 +1,9 @@
+@echo on
+
+cmake %CMAKE_ARGS% -GNinja %SRC_DIR% -DRust_CARGO_TARGET=%CARGO_BUILD_TARGET%
```
Comment (jjerphan) [inline recipes/resolvo-cpp/bld.bat:3]: This makes the build fail (my bad).
```suggestion
cmake %CMAKE_ARGS% -GNinja %SRC_DIR%
```

**Review by jjerphan — APPROVED** [review]
LGTM from a user and a feedstock maintainer perspectives!

Thank you, @baszalmstra.

I have been able to build and use it locally after a local installation. I think this is a good first version for those bindings; we can think of publishing the static library later, when we strictly require it.

File: recipes/resolvo-cpp/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoTargets.cmake  # [unix]
+    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoConfigVersion.cmake  # [unix]
+    - test -e ${PREFIX}/lib/cmake/Resolvo/ResolvoConfig.cmake  # [unix]
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\include\resolvo\resolvo_internal.h (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\resolvo_cpp.dll.lib (exit 1)  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoTargets.cmake  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfigVersion.cmake  # [win]
+    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfig.cmake  # [win]
```
Comment (jjerphan) [inline recipes/resolvo-cpp/meta.yaml:41]:
```suggestion
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoTargets.cmake  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfigVersion.cmake  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfig.cmake  # [win]
```
```suggestion
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoTargets.cmake (exit 1)  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfigVersion.cmake (exit 1)  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\Resolvo\ResolvoConfig.cmake (exit 1)  # [win]
```

**Comment by SylvainCorlay** [comment]
Thank you this is amazing!

**Pull request merged**